File: wfm-develop mobile 07.05.2026/wfm/mobile/android/app/src/main/java/com/beyondviolet/wfm/deeplink/DeepLinkHandler.kt
package com.beyondviolet.wfm.deeplink

import android.net.Uri

/**
 * Обработчик deep link для авторизации
 */
object DeepLinkHandler {

    /**
     * Извлечь код авторизации из deep link
     * Формат: wfm://auth/code_reciver/{code}/
     *
     * @param uri Deep link URI
     * @return 4-значный код или null если формат неверный
     */
    fun extractAuthCode(uri: Uri?): String? {
        if (uri == null) return null

        // Проверяем scheme и host
        if (uri.scheme != "wfm" || uri.host != "auth") return null

        // Получаем path segments: ["code_reciver", "1234"]
        val pathSegments = uri.pathSegments
        if (pathSegments.size < 2 || pathSegments[0] != "code_reciver") return null

        val code = pathSegments[1]

        // Валидация: только 4 цифры
        if (code.length != 4 || !code.all { it.isDigit() }) return null

        return code
    }
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMAuth/Sources/WFMAuth/Utilities/DeepLinkHandler.swift
import Foundation

/// Обработчик deep link для авторизации
public struct DeepLinkHandler {

    /// Извлечь код авторизации из deep link
    /// Формат: wfm://auth/code_reciver/{code}/
    ///
    /// - Parameter url: Deep link URL
    /// - Returns: 4-значный код или nil если формат неверный
    public static func extractAuthCode(from url: URL) -> String? {
        // Проверяем scheme
        guard url.scheme == "wfm" else { return nil }

        // Проверяем host
        guard url.host == "auth" else { return nil }

        // Получаем path segments: ["code_reciver", "1234"]
        let pathComponents = url.pathComponents.filter { $0 != "/" }

        guard pathComponents.count >= 2,
              pathComponents[0] == "code_reciver" else { return nil }

        let code = pathComponents[1]

        // Валидация: только 4 цифры
        guard code.count == 4,
              code.allSatisfy({ $0.isNumber }) else { return nil }

        return code
    }
}
